### ISSUE TITLE
Help needed writing a rule for markdown-it

### ISSUE THREAD

**Issue by Rishgod** [markdown-it/markdown-it#648]
Hey, I was working in another Open source project 'Joplin' which uses markdown it, so there was a feature request the to have a button to copy the code to the clipboard for every fenced token type,

Something like this 
![copy-to-clipboard](https://user-images.githubusercontent.com/54576074/76708208-15752c80-671b-11ea-9d4e-f20e2395146a.gif)

So I did also found a great library Clipboard.js to implement the same, it could be done by

```
<button class='clipboardjs' data-clipboard-target='#my-textarea'>Click to copy</button>
```

so basically to implement this I `data-clipboard-target = token[idx].context ` for the fenced rule, so can you please guide me in how can I get access to tokens of all fence type,

I did read the docs and developer guides, I tried to understand them but still there are a bit if gaps in my knowledge.

Regards, 
Rishabh.

**Comment by puzrin** [comment]
Sorry, no resources to help with generic programming.

**Comment by mouday** [comment]
how do you implement?
